src/test/store/actions.test.js

```diff
@@ -384,3 +387,111 @@ describe('actions/updateProfileSelection', function() {
     });
   });
 });
+
+describe('app actions', function() {
+  describe('changeSelectedTab', function() {
+    it('can change tabs', function() {
+      const { dispatch, getState } = storeWithSimpleProfile();
+      expect(UrlStateSelectors.getSelectedTab(getState())).toEqual('calltree');
+      dispatch(AppActions.changeSelectedTab('stack-chart'));
+      expect(UrlStateSelectors.getSelectedTab(getState())).toEqual(
+        'stack-chart'
+      );
+    });
+  });
+
+  describe('profilePublished', function() {
+    const hash = 'de0ecf545819a95caa4c4d55b7a9cedb';
+    it('changes the data source to public', function() {
+      const { dispatch, getState } = storeWithSimpleProfile();
+      expect(UrlStateSelectors.getDataSource(getState())).toBe('none');
+      dispatch(AppActions.profilePublished(hash));
+      expect(UrlStateSelectors.getDataSource(getState())).toBe('public');
+    });
+
+    it('changes the data source to public', function() {
+      const { dispatch, getState } = storeWithSimpleProfile();
+      expect(UrlStateSelectors.getHash(getState())).toBe('');
+      dispatch(AppActions.profilePublished(hash));
+      expect(UrlStateSelectors.getHash(getState())).toBe(hash);
+    });
+  });
+
+  describe('changeTabOrder', function() {
+    it('can change the saved tab order', function() {
+      const { dispatch, getState } = storeWithSimpleProfile();
+      expect(ProfileViewSelectors.getTabOrder(getState())).toEqual([
+        0,
+        1,
+        2,
+        3,
+      ]);
+      dispatch(AppActions.changeTabOrder([2, 3, 1, 0]));
+      expect(ProfileViewSelectors.getTabOrder(getState())).toEqual([
+        2,
+        3,
+        1,
+        0,
+      ]);
+    });
+  });
+
+  describe('urlSetupDone', function() {
+    it('will remember when url setup is done', function() {
+      const { dispatch, getState } = storeWithSimpleProfile();
+      expect(AppSelectors.getIsUrlSetupDone(getState())).toEqual(false);
+      dispatch(AppActions.urlSetupDone());
+      expect(AppSelectors.getIsUrlSetupDone(getState())).toEqual(true);
```

Comment:
can you try to test for sendAnalytics too? I think that you can add a "ga" var to window using sinon or jest and then check it's been called.

Reply:
I created a `withAnalyticsMock`, mainly because I'm adverse to side-effects. It will provide the list of analytics and clean up after itself.